networkd/network_server.ml

```diff
@@ -89,6 +89,22 @@ let set_dns_interface _ dbg ~name =
  * to all incoming packets, which confuses certain guests OSes. The workaround
  * constitutes adding a VLAN0 Linux device to strip those headers again.
  *)
+
+let is_vlan_in_use _ dbg ~vlan =
+	(* Identify the vlan is used by FCoE which is unknown to XAPI *)
+	Debug.with_thread_associated dbg (fun () ->
+		let vlan_dir = "/proc/net/vlan" in
+		let vlans =
+			try
```

Comment:
Could be written like this - it's a matter of taste.

    let vlan_dir = "/proc/net/vlan" in
    try
      Sys.readdir vlan_dir
      |> Array.to_list
      |> List.map (fun vlan -> String.split '-' vlan |> List.hd)
      |> List.mem vlan
    with
      e -> error "can't read %s." vlan_dir